cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/AbstractProgramRuntimeService.java

```diff
@@ -177,14 +276,14 @@ protected void updateRuntimeInfo(ProgramType type, RunId runId, RuntimeInfo runt
     lock.lock();
     try {
       if (!runtimeInfos.contains(type, runId)) {
-        runtimeInfos.put(type, runId, programStarted(runtimeInfo));
+        runtimeInfos.put(type, runId, programStarted(runtimeInfo, null));
```

Comment:
It's better not to use null (and target methods are not `@Nullable` anyway). Use a no-op Runnable.